### DISCUSSION TITLE
unable to ssh with github actions token of teleport.

### DISCUSSION THREAD

**Discussion by hdavila146** [gravitational/teleport#46116] [Q&A]
Expected behavior:

Current behavior:

Bug details:
- Teleport version 16.1.7
- Recreation steps
using a github action to deploy a playbook of ansible 
`- name: Install Teleport
        uses: teleport-actions/setup@v1
        with:
          version: 16.1.4
      
      - name: Fetch credentials using Machine ID
        id: auth
        uses: teleport-actions/auth@v2
        with:
          # Use the address of the auth/proxy server for your own cluster.
          proxy: teleport.domain.com:443
          # Use the name of the join token resource you created in step 1.
          token: ansbile-token
          # Specify the length of time that the generated credentials should be
          # valid for. This is optional and defaults to "1h"
          certificate-ttl: 1h
          # Enable the submission of anonymous usage telemetry.
          anonymous-telemetry: 1

      - name: List nodes
        run: tsh -i ${{ steps.auth.outputs.identity-file }} ls
     
      - name: Install Ansible 
        shell: bash
        run: |
          sudo apt update
          sudo apt install -y software-properties-common
          sudo add-apt-repository --yes --update ppa:ansible/ansible
          sudo apt install -y ansible
    
      - name: Test Ansible 
        run: |
          tsh config > ssh.cfg
          tsh ls --format=json | jq '.[].spec.hostname + ".teleport.domain.com"' > hosts
          variations 
          ssh -vvv -F ./ssh.cfg root@host "echo foobar"
          **ssh: Could not resolve hostname test-bench-1: Temporary failure in name resolution
          Error: Process completed with exit code 255.**
          ssh -vvv -F ./ssh.cfg root@host.teleport.domain.com "echo foobar"
          **Host key verification failed.
          Error: Process completed with exit code 255.**
        - Debug logs
        ssh -vvv -F ./ssh.cfg root@host
          **ssh: Could not resolve hostname test-bench-1: Temporary failure in name resolution
          Error: Process completed with exit code 255.**
          ssh -vvv -F ./ssh.cfg root@host.teleport.domain.com
          **Host key verification failed.
          Error: Process completed with exit code 255.**

**Comment by hdavila146** [comment]
Does anyone have a similar issue using the teleport token and connecting to a node and use ansible playbook? thanks?

**Comment by webvictim** [comment]
You'd need to make reference to the outputted SSH config file in your commands - see if you can get this to work:

```yaml
- name: Run hostname using SSH (OpenSSH)
  run: ssh -F ${{ steps.auth.outputs.ssh-config }} root@example-node.example.teleport.sh hostname
```

You can also use `tsh ls` with the outputted identity file to get a list of hosts:

```yaml
- name: Get Ansible hosts
  run: |
    tsh -i ${{ steps.auth.outputs.identity-file }} ls --format=json | jq '.[].spec.hostname + ".teleport.domain.com"' > hosts
```

Putting these together should get you nearer to where you need to be.

See these guides for a little more detail:
- https://goteleport.com/docs/enroll-resources/machine-id/deployment/github-actions/#example-teleport-actionsauth
- https://goteleport.com/docs/enroll-resources/machine-id/access-guides/ansible/
